pkg/health/heartbeat/heartbeat.go

```diff
@@ -0,0 +1,277 @@
+func (hb *Heartbeat) RenewMemberLeases() error {
+	//Fetch lease associated with member
+	memberName, err := GetEnvVarOrError("POD_NAME")
+	fmt.Println("member name is ", memberName)
+	if err != nil {
+		return &errors.EtcdError{
+			Message: fmt.Sprintf("POD NAME env var not present: %v", err),
+		}
+	}
+	namespace, err := GetEnvVarOrError("POD_NAMESPACE")
+	if err != nil {
+		return &errors.EtcdError{
+			Message: fmt.Sprintf("POD_NAMESPACE env var not present: %v", err),
+		}
+	}
+	memberLease := &v1.Lease{}
+	err = hb.K8sClient.Get(context.TODO(), client.ObjectKey{
+		Namespace: namespace,
+		Name:      memberName,
+	}, memberLease)
+	if err != nil {
+		return &errors.EtcdError{
+			Message: fmt.Sprintf("Could not fetch member lease: %v", err),
+		}
+	}
+
+	//Create etcd client to get etcd ID
+	etcdClient, err := etcdutil.GetTLSClientForEtcd(hb.etcdConfig)
+	if err != nil {
+		return &errors.EtcdError{
+			Message: fmt.Sprintf("Failed to create etcd client: %v", err),
+		}
+	}
+	defer etcdClient.Close()
+	response, err := etcdClient.Status(context.TODO(), hb.etcdConfig.Endpoints[0])
+	if err != nil {
+		return &errors.EtcdError{
+			Message: fmt.Sprintf("Failed to get status of etcd endPoint [ %v ] with error: %v", hb.etcdConfig.Endpoints[0], err),
+		}
+	}
+	memberID := ""
+	if response.Header.MemberId == response.Leader {
+		memberID = strconv.FormatUint(response.Header.MemberId, 16) + ":Leader"
+	} else {
+		memberID = strconv.FormatUint(response.Header.MemberId, 16) + ":Member"
+	}
+
+	//Change HolderIdentity and RenewTime of lease
+	renewedMemberLease := memberLease.DeepCopy()
+	actor := memberID
+	renewedMemberLease.Spec.HolderIdentity = &actor
+	renewedTime := time.Now()
+	renewedMemberLease.Spec.RenewTime = &metav1.MicroTime{Time: renewedTime}
+	renewedMemberLease.Spec.LeaseDurationSeconds = &hb.leaseDuration
+
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		return hb.K8sClient.Patch(context.Background(), renewedMemberLease, client.MergeFrom(memberLease))
+	})
+	if err != nil {
+		return &errors.EtcdError{
+			Message: fmt.Sprintf("Failed to renew member lease: %v", err),
+		}
+	}
+
+	hb.logger.Info("Renewed member lease for etcd at ", renewedTime)
+
+	return nil
+}
+
+// UpdateFullSnapshotLeases updates the holderIdentity field of the full snapshot lease with the last revision in the latest full snapshot
+func UpdateFullSnapshotLeases(logger *logrus.Entry, fullSnapshot *brtypes.Snapshot, k8sClientset client.Client) error {
+	if fullSnapshot == nil {
+		return &errors.EtcdError{
+			Message: "can not update full snapshot lease, nil snapshot passed",
+		}
+	}
+
+	namespace, err := GetEnvVarOrError("POD_NAMESPACE")
+	if err != nil {
+		return &errors.EtcdError{
+			Message: fmt.Sprintf("Pod namespace env var not present: %v", err),
+		}
+	}
+
+	if err != nil {
+		return &errors.EtcdError{
+			Message: fmt.Sprintf("Failed to create k8s clientset: %v", err),
+		}
+	}
```

Comment:
if `err!=nil` then you are already returning the error above(line no 150 to 154) then this is unreachable.

Reply:
Thanks for catching this. Removed unreachable code